docs/design/rubygems.md

```diff
@@ -0,0 +1,425 @@
+
+#### Plugins
+
+Not supported by cachi2.
+
+### Platforms
+
+Some gems may contain pre-compiled binaries that provide native extensions to the Ruby package.
+One of the goals of cachi2 is to enforce building from source as much as possible.
+([pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats) are an exception)
+
+To satisfy this goal, we need some way of avoiding dependencies that contain binaries.
+This can be achieved through the `BUNDLE_FORCE_RUBY_PLATFORM` environment variable.
+See environment variables [section](#environment-variables).
+
+For example - all versions (platforms) of nokogiri gem:
+
+<https://rubygems.org/gems/nokogiri/versions/>
+
+### Checksums
+
+Checksum validation is enabled by default.
+It can be disabled with the `BUNDLE_DISABLE_CHECKSUM_VALIDATION` environment variable.
+
+There is also an option to generate checksums in `Gemfile.lock`, but in bizarre way.
+This feature is not exposed to users right now.
+Checksums are not generated by default, but they can be added manually.
+There is an [issue](https://github.com/rubygems/rubygems/issues/3379#issuecomment-1974761996)
+in the official rubygems repository as well.
+
+```shell
+# manually add `CHECKSUMS` section somewhere in the Gemfile.lock
+vim Gemfile.lock
+# install any gem
+bundle add rails --version "6.1.7"
+# check the Gemfile.lock /o\
+cat Gemfile.lock
+```
+
+Example of a checksum section in the `Gemfile.lock`:
+
+```Gemfile.lock
+...
+DEPENDENCIES
+ rails (= 6.1.7)
+
+CHECKSUMS
+ actioncable (6.1.7) sha256=ee5345e1ac0a9ec24af8d21d46d6e8d85dd76b28b14ab60929c2da3e7d5bfe64
+ actionmailbox (6.1.7) sha256=c4364381e724b39eee3381e6eb3fdc80f121ac9a53dea3fd9ef687a9040b8a08
+ actionmailer (6.1.7) sha256=5561c298a13e6d43eb71098be366f59be51470358e6e6e49ebaaf43502906fa4
+ actionpack (6.1.7) sha256=3a8580e3721757371328906f953b332d5c95bd56a1e4f344b3fee5d55dc1cf37
+ actiontext (6.1.7) sha256=c5d3af4168619923d0ff661207215face3e03f7a04c083b5d347f190f639798e
+ actionview (6.1.7) sha256=c166e890d2933ffbb6eb2a2eac1b54f03890e33b8b7269503af848db88afc8d4
+ ...
+
+BUNDLED WITH
+ 2.5.11
+```
+
+I believe this feature is available since Bundler [v2.5.0](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/lockfile_parser.rb#L55)
+from this [PR](https://github.com/rubygems/rubygems/pull/6374) being merged on Oct 21, 2023.
+
+### Environment variables
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except BUNLDE_APP_CONFIG),
+we need to set the Bundler configuration options to make the build work.
+We can easily set the environment variables if the local configuration file does not exist.
+
+#### Relevant environment variables
```

Comment:
Do we also need to set BUNDLE_NO_PRUNE=true during the build for the multiple packages scenario to prevent one ```bundle install``` command from pruning the cache?

Reply:
I added this variable as part of the necessary variables for build in the design.